model/task/task.go

```diff
@@ -775,6 +778,39 @@ func (t *Task) AllDependenciesSatisfied(cache map[string]Task) (bool, error) {
 	return true, nil
 }
 
+// MarkDependenciesFinished updates all direct dependencies on this task to
+// cache whether or not this task has finished running.
+func (t *Task) MarkDependenciesFinished(finished bool) error {
```

Comment:
Can we maybe call this function SetDependenciesFinished? I think that may make it clearer that we are setting a db field as opposed to marking finished as true.

Reply:
Evergreen is inconsistent about this because Set, Mark, and Update are all used interchangeably when updating something in the DB. For example, `MarkUnattainableDependencies` is extremely similar to this function. I would prefer to leave it alone since it's consistent with the other dependency-modifying functions.